src/components/Map.tsx

```diff
@@ -163,6 +167,28 @@ const Map = ({
         zoomEnabled={interactive}
         compassEnabled={false}
       >
+        {(showUserLocation || currentLocationEnabled) && (
+          <MapboxGL.UserLocation
+            onUpdate={(loc) => {
```

Comment:
given that this callback gets called super frequently, we might want to move it into a defined const function with `useCallback`